bot/extensions/clashofcode/views.py

```diff
@@ -0,0 +1,160 @@
+import asyncio
+import logging
+import time
+
+import discord
+from codingame.http import HTTPError
+from discord import ui
+
+from bot import core
+from bot.config import settings
+from bot.extensions.clashofcode.utils import coc_client, coc_helper, languages, modes
+
+
+def em(mode, players):
+    embed = discord.Embed(title="**Clash started**")
+    embed.add_field(name="Mode", value=mode, inline=False)
+    embed.add_field(name="Players", value=players)
+    return embed
+
+
+class CreateCocView(ui.View):
+    LANGUAGES_CUSTOM_ID = "extensions:clashofcode:languages"
+    MODES_CUSTOM_ID = "extensions:clashofcode:modes"
+    SELECT_ALL_CUSTOM_ID = "extensions:clashofcode:select_all"
+    DESELECT_ALL_CUSTOM_ID = "extensions:clashofcode:deselect_all"
+    CREATE_CUSTOM_ID = "extensions:clashofcode:create"
+
+    @discord.ui.select(
+        custom_id=LANGUAGES_CUSTOM_ID,
+        placeholder="➕ Select a language",
+        options=[discord.SelectOption(label=lang, default=True) for lang in languages],
+        max_values=len(languages),
+    )
+    async def select_languages(self, interaction: core.InteractionType, _select: ui.Select):
+        await interaction.response.defer()
+
+    @discord.ui.button(label="Select All", style=discord.ButtonStyle.blurple, emoji="🟢", custom_id=SELECT_ALL_CUSTOM_ID)
+    async def select_all_languages(self, interaction: core.InteractionType, _button: ui.Button):
```

Comment:
`select_all` or `select_all_modes` would be better because it selects all the modes and doesn't do anything to languages

Reply:
huh? It selects all languages, did you test this at all or did it not work for you.